File: finputSelectionSort.py
# Selection Sort

def selectionsort(data):
    n = len(data)
    for i in range(0, n - 1):
        si = i
        for j in range(i + 1, n):
            if data[j] < data[si]:
                si = j
        data[i], data[si] = data[si], data[i]
    return data

def main():
    n = int(input("\nEnter the number of elements to be sorted: "))
    raw_data = []
    
    print("\nEnter the elements to be sorted:")
    for _ in range(n):
        raw_data.append(float(input()))
    
    selectionsort(raw_data)
    
    print("\nThe sorted elements are:")
    for num in raw_data:
        print(f"{num:4.2f}", end=' ')
    print("\n")

if __name__ == "__main__":
    main()

File: finputSelectionSort.c
#include <stdio.h>

void selectionsort(float data[], int n){
    for (int i = 0; i < n-1; i++){
        int si = i;
        for (int j = i+1; j < n; j++){
            if (data[j] < data[si]){
                si = j;
            }
        }
        float temp = data[i];
        data[i] = data[si];
        data[si] = temp;
    }
}


int main()
{
    float raw_data[20];
    int n, i;

    printf("\nEnter the number of elements to be sorted: ");
    scanf("%d", &n);

    printf("\nEnter the elements to be sorted: ");
    for(i = 0; i < n; i++)
    {
        scanf("%f", &raw_data[i]);
    }

    selectionsort(raw_data, n);

    printf("\nThe sorted elements are:\n");
    for(i = 0; i < n; i++)
    {
        printf("%4.2f ", raw_data[i]);
    }
    printf("\n");

    return 0;
}